src/utils/private_candidate.c

```diff
@@ -0,0 +1,1190 @@
+        break;
+    case SR_PC_PREFER_CANDIDATE:
+        /*
+         * Remove conflicting nodes from diff, This ensures that the corresponding
+         * data from the private candidate is preserved
+         */
+        if ((err_info = pc_unlink_conflicting_nodes(&privcand_vs_running, conflict_set))) {
+            goto cleanup;
+        }
+
+        /* merge nodes that are not in conflict */
+        if ((err_info = sr_lyd_diff_apply_all(&privcand_ds_tree, privcand_vs_running))) {
+            goto cleanup;
+        }
+
+        /* reset private candidate to match current running datastore */
+        lyd_free_all(privcand->diff_backup);
+        privcand->diff_backup = NULL;
+
+        lyd_free_all(privcand->diff_privcand);
+        privcand->diff_privcand = NULL;
+
+        /* The current running is now considered the baseline for the private candidate datastore.
+         * Changes are stored in diff_privcand */
+        if ((err_info = sr_lyd_diff_siblings(running_ds_tree, privcand_ds_tree, 0, NULL, &privcand->diff_privcand))) {
+            goto cleanup;
+        }
+
+        sr_pc_free_conflicts(*conflict_set);
+        (*conflict_set) = NULL;
+        break;
+    default:
+        /* should never happen */
+        assert(0);
+        break;
+    }
+
+cleanup:
+    lyd_free_all(running_vs_privcand);
+    lyd_free_all(final_diff);
+    lyd_free_all(privcand_vs_running);
+
+    return err_info;
+}
+
+/**
+ * @brief Store a conflict between candidate and running datastore nodes.
+ *
+ * After resolving the conflicts, the private candidate's internal diff is rebuilt to reflect the resolved state.
+ *
+ * @param[in] cand_iter Pointer to the conflicting node from the private candidate datastore.
+ * @param[in] run_iter Pointer to the conflicting node from the running datastore.
+ * @param[in] type The type of conflict.
+ * @param[in] dup Wheter to duplicate @p run_iter and @p cand_iter into conflicts.
+ *                If zero, the function will assign pointers without duplication.
+ * @param[out] conflict_set List of conflicts which are not resolved yet.
+ * @return err_info, NULL on success.
+ */
+static sr_error_info_t *
+pc_add_conflict(struct lyd_node *cand_iter, struct lyd_node *run_iter, sr_pc_conflict_type_t type, int dup, sr_pc_conflict_set_t **conflict_set)
+{
+    sr_error_info_t *err_info = NULL;
+    sr_pc_conflict_info_t *c_info;
+
+    if (conflict_set && !(*conflict_set)) {
+        *conflict_set = calloc(1, sizeof(**conflict_set));
+        SR_CHECK_MEM_GOTO(!(*conflict_set), err_info, cleanup);
+    }
+
+    // (*conflict_set)->conflicts = sr_realloc((*conflict_set)->conflicts, ((*conflict_set)->conflict_count + 1) * sizeof *(*conflict_set)->conflicts);
+    // SR_CHECK_MEM_GOTO(!(*conflict_set)->conflicts, err_info, cleanup);
+
+    /* allocate memory for new conflicts */
+    sr_pc_conflict_info_t *tmp = realloc((*conflict_set)->conflicts, ((*conflict_set)->conflict_count + 1) * sizeof *tmp);
```

Comment:
Never declare variables elsewhere than on the beginning of a block. Also, use `void *` type for these purposes.